### ISSUE TITLE
simplify bootstrap code

### ISSUE THREAD

**Issue by kevinushey** [rstudio/renv#890]
Right now, `renv` injects a pretty large subset of code into `renv/activate.R` to try and bootstrap itself. We should try to minimize some of this if possible.

In the simplest scenario, we could try to load `renv` from the user library, and then use that to load the associated project.

**Comment by hadley** [comment]
Are you imaging that `template-activate.R` becomes something very roughly like this:

```R
if (requireNamespace("renv", quietly = TRUE)) {
  templib <- tempfile()
  dir.create(templib)

  install.packages("renv", templib)
  if (packageVersion("renv", templib) < "0.18") {
    stop("Failed to find new enough renv")
  }
  loadNamespace("renv", lib.loc = templib)
}

renv:::renv_bootstrap("${VERSION}")
```

That code assumes installing renv from CRAN (or other preconfigured `repos`), but maybe it would make sense to provide a renv only repo? (That could either be used exclusively or as fallback).

**Comment by kevinushey** [comment]
Yeah, exactly. The other main things we might need to support would be:

- Support `RENV_BOOTSTRAP_TARBALL` (https://github.com/rstudio/renv/blob/8b145e8856f3a9e782dd51662b3247662067aa11/R/bootstrap.R#L308-L338)

- Support the various ways that repositories might be set. (I sort of regret allowing so many disparate ways for these to be provided now...)

- Think about whether we still need to support custom download headers in bootstrap.

We might also need to think about supporting the repositories as defined in the `renv.lock` lockfile?

**Comment by hadley** [comment]
So at least for now we probably need to preserve `renv_bootstrap_repos()`, and `renv_bootstrap_download_tarball()`. 

I'd suggest doing this in two steps: we first create `renv_bootstrap()` in `bootstrap2.R` and then move in all the `renv_bootstrap_*` functions that are no longer needed in `bootstrap.R`. Then we can reconsider all the `renv_bootstrap()` functions and if they can be simplified by using other renv tools that were previously out of scope.

We'll also need to solve some meta-bootstrapping problem, in that we'll want to use the new activate script before renv is available on CRAN. I think that is probably most easily solved with a custom repo? Maybe we could use r-universe.dev?

**Comment by kevinushey** [comment]
That sounds like the right approach to me.
